packages/compiler-cli/test/ngtsc/ngtsc_spec.ts

```diff
@@ -357,6 +357,147 @@ runInEachFileSystem(os => {
         const jsContents = env.getContents('test.js');
         expect(jsContents).toContain('/** @nocollapse */ TestCmp.ɵcmp');
       });
+
+      /**
+       * The following set of tests verify that after Tsickle run we do not have cases which trigger
+       * automatic semicolon insertion, which breaks the code. In order to avoid the problem, we
+       * wrap all function expressions in certain fields ("providers" and "viewProviders") in
+       * parentheses. More info on Tsickle processing related to this case can be found here:
+       * https://github.com/angular/tsickle/blob/d7974262571c8a17d684e5ba07680e1b1993afdd/src/jsdoc_transformer.ts#L1021
+       */
+      describe(
+          'wrap functions in certain fields in parentheses when closure annotations are requested',
+          () => {
+            const providers = `
+              [{
+                provide: 'token-a',
+                useFactory: (service: Service) => {
+                  return () => service.id;
+                }
+              }, {
+                provide: 'token-b',
+                useFactory: function(service: Service) {
+                  return function() {
+                    return service.id;
+                  }
+                }
+              }]
+            `;
+
+            const service = `
+              export class Service {
+                id: string = 'service-id';
+              }
+            `;
+
+            const verifyOutput = (jsContents: string) => {
+              // verify that there is no pattern that triggers automatic semicolon insertion
```

Comment:
Perhaps add a bit more explanation?

```suggestion
              // verify that there is no pattern that triggers automatic semicolon insertion
              // by checking that there are no return statements not wrapped in parentheses.
```